napps/kytos/of_core/main.py

```diff
@@ -145,53 +147,101 @@ def handle_echo_request(self, event):
         """
         log.debug("Echo Request message read")
 
-        echo_request = event.content['message']
+        echo_request = event.message
         echo_reply = EchoReply(xid=echo_request.header.xid)
         event_out = KytosEvent(name=('kytos/of_core.messages.out.'
                                      'ofpt_echo_reply'),
                                content={'message': echo_reply,
                                         'destination': event.source})
         self.controller.buffers.msg_out.put(event_out)
 
+    @listen_to('kytos/core.connection.new')
+    def handle_core_new_connection(self, event):
+        """Method called when a new switch is connected.
+
+        This method will send a hello world message when a new switch is
+        connected.
+        """
+        self.say_hello(event.source)
```

Comment:
Hehe not ridiculous at all.


	Switch:
		- Hi, I can talk v0x02. [Hello version 0x02, xid=whatever]
	Controller:
		- Hi, I can talk v0x01. [Hello version 0x01, xid=whatever]

	When each side receives the Hello:
	Switch:
		(Hum, this guy only talks 0x01, better talk 0x01).
		Sets the version for the connection and wait for messages.
		'Hello Complete!'
	Controller:
		(Hum, this guy already knows 0x02, he must know 0x01. Let's talk 0x01)
		Sets the version for the connection;
		'Hello Complete!'
		and start a conversation (maybe send a Features Request).
		

This poses a problem when one of the parts does not implement all versions up to
their greater version (as is the case with Kytos). That's why the negotiation
changed in 1.3.1, and now carries a version bitmap. Problem example (switch
only talks v0x01 and v0x03, and controller v0x01 and v0x02):

	Switch:
		- Hi, I can talk v0x03. [Hello version 0x03, xid=whatever]
	Controller:
		- Hi, I can talk v0x02. [Hello version 0x02, xid=whatever]

	When each side receives the Hello:
	Switch:
		(Hum, this guy only talks 0x02, better talk 0x02, but I don't know it!).
		Hello Failed!
	Controller:
		(Hum, this guy already knows 0x03, he must know 0x02. Let's talk 0x02)
		Sets the version for the connection and start a conversation (maybe
		send a Features Request).
		Hello Complete!

	Controller receives the hello failed, and connection is closed.

They never get to talk, although they could be talking v0x01.
Version bitmap from OF1.3.1 fixes this (but only for those talking at least
OF1.3.1)